### DISCUSSION TITLE
Translation to Ukrainian

### DISCUSSION THREAD

**Discussion by evd316** [kitadai31/revanced-patches-android6-7#28] [General]
If you need (and I see that you need to) correct errors / typos in the Ukrainian and Russian translations of the menu, then I can help. I also see that punctuation needs to be corrected.
You fixed the 'Shorts in Subscriptions' bug so quickly that I wanted to help.
![Screenshot_20230502-060500](https://user-images.githubusercontent.com/65615541/236202008-46e559dd-dd2b-47b6-ac70-49045c27ecaa.png)

**Comment by kitadai31** [comment]
Thank you!
If you send me a pull request, I will reflect it.

Or, tell me the sentence that should be corrected here.

https://github.com/kitadai31/revanced-patches-android6-7/blob/revanced-extended/src/main/resources/youtube/translations/ru-rRU/strings.xml
https://github.com/kitadai31/revanced-patches-android6-7/blob/revanced-extended/src/main/resources/youtube/translations/uk-rUA/strings.xml

Translasions file is these.
If you write the line number and the new translation here, I will reflect it.

**Reply by evd316** [reply]
What is 'pull request'?
Strings UKR and RUS do not match?

I'll try uk-rUA first. I'm not a philologist, but my spelling UKR/RUS is very good). I will not change the UKR translation itself, only what "hurts the eye".
In my opinion, the RUS translation is made better in terms of meaning.

**Comment by evd316** [comment]
uk-rUA:
string 076 Вилучення картки альбому
string 139 Вилучення оповіщення Містить пряму рекламу (у відео) вимкнено
string 140 Вилучення оповіщення Містить пряму рекламу (у відео) увімкнено
string 141 Містить пряму рекламу (у відео)
string 168 Типова якість відео для моб. мережі
string 169 Типова якість відео для моб. мережі
string 181 Вимкнути вібровідгук при перемотуванні жестом вгору по рядку перемотування
string 184 Вимкнути вібровідгук при перемотуванні довгим натисканням
string 187 Вимкнути вібровідгук при зміні глави
string 192 Вимкнення вібровідгуку при збільшенні для заповнення екрана
string 202 Увімкнути додаткові параметри швидкості відео
string 203 Відкриття посилань через \nзовнішній браузер (Android WebView)\n(Натисніть, щоб змінити параметр)
string 205 Внутрішній браузер (Android WebView)
string 215 Перенаправлення посилань через youtube.com/redirect вимкнено\n(Працює лише в описі відео)
string 216 Перенаправлення посилань через youtube.com/redirect увімкнено\n(Працює лише в описі відео)
string 220 Press-to-Swipe (Дотик для жесту)
string 232 Автояскравість не активується при яскравості, встановленій жестом на 0
string 233 Автояскравість активується при яскравості, встановленій жестом на 0
string 354 Меню зворотнього зв'язку показано
string 355 Меню зворотнього зв'язку приховано
string 360 Меню "Слухати через YouTube Music" відображається
string 361 Меню "Слухати через YouTube Music" приховано
string 362 Приховати меню "Слухати через YouTube Music"
string 378 Показувати "Мікс плейлист"
string 379 "Мікс плейлист" приховано
string 380 Приховати "Мікс плейлист"
string 465 Параметрі приховування компонентів опції спливаючої панелі плеєра
string 466 Компонент спливаючої панелі плеєра
string 536 Перед відправкою будь-якого сегмента рекомендується прочитати інструкції SponsorBlock
string 547 Жарти без контексту
string 571 Пропущено закінчення

The logic of how the function works is not clear:
string 218 Керування жестами завжди увімкнено disabled (there are two opposing concepts, I suggest […увімкнено disabled] replace with [Керування жестами завжди вимкнено])
string 219 Керування жестами вимагає тривалого натискання перед використанням enabled (I suggest […використанням enabled] replace with [Керування жестами вимагає тривалого натискання перед використанням увімкнено])

Swapped in places (function checked by me in the settings):
string 429 Смугу для перемотування та час перегляду увімкнено
string 430 Смугу для перемотування та час перегляду вимкнено

I don't understand (but maybe you need so?): to do — find and replace
all strings if exist necessary \quotes are followed by a backslash [\"] (50 entries) и \apostrophe [\'] (11 entries) — I suggest everywhere without a backslash ["] и [']
all strings if exist ненужные remove quotes at the beginning (25 strings) and end (27 strings) of sentences

**Comment by kitadai31** [comment]
I'm sorry for the late reply.
Thank you!
I will reflect it in the next update.

And I will put your name in the change history.

---

\> quotes at the beginning (25 strings) and end (27 strings) of sentences

Double quotes before and after the string are necessary to reflect line breaks.

https://developer.android.com/guide/topics/resources/string-resource#escaping_quotes
\> You can avoid this by enclosing the relevant part of your string in double quotes. In this case all whitespace characters (including new lines) will get preserved within the quoted region.

See also the original English file.
https://github.com/kitadai31/revanced-patches-android6-7/blob/revanced-extended/src/main/resources/youtube/settings/host/values/strings.xml

**Reply by evd316** [reply]
Oh thanks for that! I didn't do much, but thank you so!

-> quotes at the beginning (25 strings) and end (27 strings) of sentences
Well, then, if you need it, then you need it. I will learn for the future.

uk-rUA:
string 465v.2 Параметри приховування компонентів спливаючої панелі плеєра
string 291 Кнопка живого чату
string 344 Кнопка живого чату

ru-rRU:
Honestly, there is no desire to check&correct russian translate yet — they are (putin’s army) still bombing my town Nikopol for 12 months. So I guess I'll hold off.

I'll try to participate in future versions too (UA only), if you don't mind.

**Reply by kitadai31** [reply]
Thanks. Take care!

**Comment by evd316** [comment]
Hi, I have written about this confusion before.

in uk-rUA translation _must be_ (function checked by me in the settings):
string 429 Смугу для перемотування та час перегляду увімкнено
string 430 Смугу для перемотування та час перегляду вимкнено

or replace triger string 429 "OFF"-to-"ON" and string 430 "ON"-to-"OFF" — just one thing.

UPD: I just noticed that you haven't made all my edits yet in 2.160.5. It's a pity.
There are still items that are not fully translated from English into Ukrainian. I could translate.
![Screenshot_20230816-202806](https://github.com/kitadai31/revanced-patches-android6-7/assets/65615541/c1c2f399-5b7d-4150-b683-e03d0ee98822)

**Reply by kitadai31** [reply]
The last update (2.160.5) was emergency, so I couldn't reflect your translation, sorry.
By the way, thank you for new information!

**Reply by evd316** [reply]
I can’t find in the UKR translation (uk-rUA/strings) the string with the menu item in the screenshot above and circled in red. Did you delete it? It seems to me that you did the right thing, since it duplicated the meaning of the next paragraph below it, and possibly the function.

**Comment by kitadai31** [comment]
Reflected in the source code. Please check diff.
https://github.com/kitadai31/revanced-patches-android6-7/commit/5e14e9314bbb6360b7a60a21528ff3c70533cae4

However, I have one question.
Is "revanced_hide_menu_help_summary_off/on" translation correct?

I checked your new translation `Меню зворотнього зв'язку показано` using Google Translation, and it says `The feedback menu is shown`.
But this setting is to hide "Help & feedback" menu.
[Screenshot_20230827-165100](https://github.com/kitadai31/revanced-patches-android6-7/assets/90122968/696e262a-a9c5-40d0-b4f7-dd00039fd42c)
Old translation was `Допомога & меню зворотного зв'язку показано` .
It was `The Help & Feedback menu is shown` and it looks like more accurate for me.

Can you check this?
I don't understand Ukrainian at all and everything is from Google Translate, so sorry if I'm wrong.

**Reply by evd316** [reply]
Agree. You're right.
"Help" is "Допомога" or "Довідка" was forgotten somehow.
I no longer remember how it was at first, but I did not translate this string from ENG, but only corrected the UKR version. But... if to translate, then:
"Feedback" can be translated as "Зворотний зв'язок" and as "Відгуки", which is shorter. As a result, we have either "Меню Допомога & Відгуки показано/приховано" or "Меню Допомога & Зворотний зв'язок показано/приховано" or "Меню Довідка & Відгуки показано/приховано" or "Меню Довідка & Зворотний зв'язок показано/приховано" at your discretion.

**Reply by kitadai31** [reply]
OK. I adopted "Меню Довідка & Відгуки показано/приховано" one, following youtube app's text.
Check the diff: https://github.com/kitadai31/revanced-patches-android6-7/commit/e2c59d9542e54b8c94bbc34e864325b48e8fe7e9

**Reply by evd316** [reply]
In this case, you need to change the string 356 to "Приховати меню Довідка & Відгуки"

**Reply by kitadai31** [reply]
Thanks. Reflected https://github.com/kitadai31/revanced-patches-android6-7/commit/f6f9c1986d64c5cf59035cf98fa3fc53f1312237

**Comment by evd316** [comment]
The word "dislike" translated to UKR as "дизлайк" (most) and "дислайк" (rarely). In strings 472-475, 479-481, 484 you have rare option. If (preferably) you indicate one, then it is better to “дизлайк” — in the word “..дизлайк..” indicate the third letter “з”.
If I'll see something incorrect in the UKR translation in future I had to write here because pull request form doesn't accept me.
![noPullRequest](https://github.com/kitadai31/revanced-patches-android6-7/assets/65615541/e96557cc-97f8-4e13-8bae-9fb3850b5265)

**Reply by kitadai31** [reply]
Thank you for considering pull request.
You have to fork this repo first. Then edit translation file and create pull request.

**Reply by evd316** [reply]
OK, I'll try.

**Reply by evd316** [reply]
It seems to have worked out. It turned out to be very convenient to change everything by myself, but I will have to wait for you to add everything to the official file for the patch or not (I hope you will add it).
If somewhere you need to reduce the number of characters, tell me string, I will select another shorter synonym word or synonymous phrase/sentence.
It would also be good to delete those strings that are no longer used in the program at all (there are no such blocks/buttons), and they just take up space like garbage.

**Comment by evd316** [comment]
Hello!
Where is this phrase located? In what file?
![зображення_viber_2024-02-14_23-21-43-118](https://github.com/kitadai31/revanced-patches-android6-7/assets/65615541/2f737656-83dc-459f-a9d7-0705e08dd11d)
It can/should be made shorter for two reasons IMHO: 1) as this is a section for advanced users, you can leave it at ENG "Copy data" aligned to the right and 2) the long UKR phrase won't let you click play/pause/FF+10s, but making a copy or 3) UKR "Копіювати дані" aligned to the right.

**Reply by kitadai31** [reply]
This is from stock app

**Comment by evd316** [comment]
Hi!!!
While you are busy (we are all waiting ;)) I used Crowdin and found that there are 16 untranslated strings in ENG file (in RUS is 22). But there are not in UKR file and maybe not in the app itself. Let me know what to do with them. I can translate them and send them to you here in Discussions. ...?
![Безымянный](https://github.com/kitadai31/revanced-patches-android6-7/assets/65615541/010376c6-33d5-4622-8447-761e7ace7d5d)

**Reply by kitadai31** [reply]
They are these strings

https://github.com/kitadai31/revanced-patches-android6-7/blob/671ab4c054bdf4e70330a9c18181842ebb000f0b/src/main/resources/youtube/settings/host/values/strings.xml#L592-L629

Feel free to translate these strings, except "evanced_spoof_player_parameter_*" strings (spoof_player_parameter patch is removed)

**Reply by evd316** [reply]
Yes, I realized that they are, but there are no corresponding lines with code in the UKR file. Where to “insert” the UKR translation? What, just copy this whole block of code (592-629) from ENG file, then paste into UKR file and translate it there? Am I allowed to “coding”?

**Reply by kitadai31** [reply]
Yes
Just add the lines

**Comment by evd316** [comment]
Done.
I want to correct (if I have time) the phrases with word “layout” there. What is this — "interface" or "mockup"?

**Reply by kitadai31** [reply]
layout is layout...
idk what word is suitable in Ukraine

**Comment by evd316** [comment]
Hi!
In ENG string 467 there are 2 spaces "…Dislike API.  Tap here…":
<string name="revanced_ryd_attribution_summary">Dislike data is provided by the Return YouTube Dislike API.  Tap here to learn more.</string>
or send a pull request?

I also notice changes in the ENG source file and UKR strings too. I have accumulated some changes on UKR translation. Should I not send a pull request and wait for a message from you that I can start translating new blocks and add some my changes on UKR translation?

**Reply by kitadai31** [reply]
Open a PR thanks

**Reply by evd316** [reply]
OK, do you mean to send a PR for the first part of my comment (about two spaces in ENG string 467)?

**Reply by kitadai31** [reply]
Any fixes do you want to fix should be a PR

**Comment by evd316** [comment]
Hi!
stats_saved_sum The "h" (год) are not translated here, but the "хв." (minutes, the dot is redundant) are translated.
And in the second screenshot, "to" (по) is not translated (can be replaced with an M/N-dash).
Where can I fix this? I can't find it in the source file.
![Screenshot_20240907-025614](https://github.com/user-attachments/assets/0bb080ef-6785-4bcd-a74b-66fc1a45c620)
![Screenshot_20240907-041855](https://github.com/user-attachments/assets/ea694a89-08cc-4f3e-8e48-884cd0fc0196)

**Reply by kitadai31** [reply]
It turns out that it's coming from integrations code and cannot translate.
https://github.com/kitadai31/revanced-integrations/blob/7d36d16f41004221b2dedea116e88bc98bd853ce/app/src/main/java/app/revanced/integrations/settingsmenu/SponsorBlockSettingsFragment.java#L39

2nd one is too.

**Reply by evd316** [reply]
From SponsorBlock or YouTube?

**Reply by kitadai31** [reply]
No, from RVX code (revanced-integrations).
The string is hard-coded in integrations, so cannot be translated.

**Reply by evd316** [reply]
I translate for SponsorBlock too and could write to them about it.
Well, if not, then no. But let's assume that I understood…))

**Reply by kitadai31** [reply]
I already sent a link
https://github.com/kitadai31/revanced-integrations/blob/7d36d16f41004221b2dedea116e88bc98bd853ce/app/src/main/java/app/revanced/integrations/settingsmenu/SponsorBlockSettingsFragment.java#L39
This line has hard-coded "h" for hours

**Reply by evd316** [reply]
Thanks! Maybe they'll fix something.
They can eventually replace it with some international symbols to avoid translation. In that first my screenshot, M-dash or N-dash, and here they can even put ":" (colon) as is customary everywhere on timelines 13:55, or leave it without translation in 5d 13h 55m format (since days also appear there as time accumulates). They will figure it out themselves, I hope)

**Reply by kitadai31** [reply]
Sorry, I do not plan to fix it.

**Reply by evd316** [reply]
"They can…", not you)
"I already sent a link" you said. I thought you sent them a link. I must have misunderstood something.

**Reply by kitadai31** [reply]
Yes
I sent *you* the link of my source code
This project is not affiliated with SponsorBlock developers at all
